component/messageHandler.go

```diff
@@ -4,82 +4,86 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"os"
+	"syscall"
 	p_buff "th2-grpc/th2_grpc_common"
 
 	"github.com/rs/zerolog/log"
 	rabbitmq "github.com/th2-net/th2-common-go/schema/modules/mqModule"
 	"github.com/th2-net/th2-common-go/schema/queue/MQcommon"
+	timestamp "google.golang.org/protobuf/types/known/timestamppb"
 )
 
 type MessageTypeListener struct {
 	MessageType    string
 	Function       func(args ...interface{})
 	RootEventID    *p_buff.EventID
 	Module         *rabbitmq.RabbitMQModule
-	AmountReceived *int
+	AmountReceived int
 	NBatches       int
 	Stats          map[string]int
+	Wait           <-chan bool
+	Ch             chan os.Signal
 }
 
-func NewListener(RootEventID *p_buff.EventID, module *rabbitmq.RabbitMQModule, BoxConf *BoxConfiguration, Function func(args ...interface{})) *MessageTypeListener {
-	amountReceived := 0
+func NewListener(RootEventID *p_buff.EventID, module *rabbitmq.RabbitMQModule, BoxConf *BoxConfiguration, wait <-chan bool, ch chan os.Signal, Function func(args ...interface{})) *MessageTypeListener {
 	return &MessageTypeListener{
 		MessageType:    BoxConf.MessageType,
 		Function:       Function,
 		RootEventID:    RootEventID,
 		Module:         module,
-		AmountReceived: &amountReceived,
+		AmountReceived: 0,
 		NBatches:       4,
+		Wait:           wait,
+		Ch:             ch,
 		Stats:          make(map[string]int),
 	}
 }
 
-func (listener MessageTypeListener) Handle(delivery *MQcommon.Delivery, batch *p_buff.MessageGroupBatch,
-	confirm *MQcommon.Confirmation) error {
+func (listener *MessageTypeListener) Handle(delivery *MQcommon.Delivery, batch *p_buff.MessageGroupBatch) error {
 
 	defer func() {
 		if r := recover(); r != nil {
 			log.Err(fmt.Errorf("%v", r)).Msg("Error occurred while processing the received message.")
+			listener.Ch <- syscall.SIGINT
+			<-listener.Wait
 		}
-		*listener.AmountReceived += 1
-		if *listener.AmountReceived%listener.NBatches == 0 {
-			log.Info().Msg("Sending Statistic Event")
+		listener.AmountReceived += 1
+		if listener.AmountReceived%listener.NBatches == 0 {
+			log.Log().Msg("Sending Statistic Event")
 			table := GetNewTable("Message Type", "Amount")
 			table.AddRow("Raw_Message", fmt.Sprint(listener.Stats["Raw"]))
 			table.AddRow("Message", fmt.Sprint(listener.Stats["Messsage"]))
-			encoded, _ := json.Marshal(table)
+			var payloads []Table
+			payloads = append(payloads, *table)
+			encoded, _ := json.Marshal(&payloads)
 			listener.Module.MqEventRouter.SendAll(CreateEventBatch(
 				listener.RootEventID, CreateEvent(
-					CreateEventID(), listener.RootEventID, GetTimestamp(), GetTimestamp(), 0, "Statistic on Batches", "message", encoded, nil),
+					CreateEventID(), listener.RootEventID, timestamp.Now(), timestamp.Now(), 0, "Statistic on Batches", "message", encoded, nil),
```

Comment:
Why we use to different time? Maybe CreateEvent should take only one timestamp argument